amberdata/src/endpoint/balance.ts

```diff
@@ -0,0 +1,122 @@
+import objectPath from 'object-path'
+import { Requester, Validator } from '@chainlink/external-adapter'
+import { AdapterRequest } from '@chainlink/types'
+import { Config, DEFAULT_DATA_PATH, getBaseURL } from '../config'
+import { CoinType, ChainType, isCoinType, isChainType } from '.'
+
+export const Name = 'balance'
+
+type Address = {
+  address: string
+  coin?: CoinType
+  chain?: ChainType
+  balance?: number
+  warning?: string
+}
+
+type RequestData = {
+  dataPath: string
+  confirmations: number
+}
+
+type ResponseWithResult = {
+  response: any
+  result: Address
+}
+
+const blockchains: { [ticker: string]: string } = {
+  btc: 'bitcoin',
+  eth: 'ethereum',
+  bch: 'bitcoin-abc',
+  ltc: 'litecoin',
+  btsv: 'bitcoin-sv',
+  zec: 'zcash',
+}
+
+const WARNING_NO_OPERATION_COIN = 'No Operation: unsupported coin'
+const WARNING_NO_OPERATION_TESTNET = "No Operation: 'testnet' is only supported on 'eth'"
+const WARNING_NO_OPERATION_CHAIN =
+  "No Operation: invalid chain parameter, must be one of 'mainnet' or 'testnet'"
+const WARNING_NO_OPERATION_MISSING_ADDRESS = 'No Operation: address param is missing'
+
+const getBalanceURI = (address: string) => `/api/v2/addresses/${address}/account-balances/latest`
+
+const getBlockchainHeader = (chain: string, coin: string) => {
+  const network = blockchains[coin]
+  if (chain === 'testnet') {
+    if (network === 'ethereum') return 'ethereum-rinkeby'
+    throw WARNING_NO_OPERATION_TESTNET
```

Comment:
This is not a warning but an error. We used warnings when processing sequences by multiple adapters, so if one adapter does not know how to process an item in the sequence, it would add a warning and pass it along.

There is an idea to add a `strict` flag, so these sequence processors would error as soon as they encounter a piece of data/request they can not process.

Reply:
Only Etherum has testnet support?

Reply:
> Only Etherum has testnet support?

Yeah, weirdly.

Reply:
> This is not a warning but an error. We used warnings when processing sequences by multiple adapters, so if one adapter does not know how to process an item in the sequence, it would add a warning and pass it along.
> 
> There is an idea to add a `strict` flag, so these sequence processors would error as soon as they encounter a piece of data/request they can not process.

We should sync about this concept further!